numpy/doc/structured_arrays.py

```diff
@@ -6,231 +6,506 @@
 Introduction
 ============
 
-NumPy provides powerful capabilities to create arrays of structured datatype.
-These arrays permit one to manipulate the data by named fields. A simple 
-example will show what is meant.: ::
+Structured arrays are ndarrays whose datatype is composed of simpler datatypes,
+which are organized as a sequence of named :term:`fields <field>`. For example, ::
 
- >>> x = np.array([(1,2.,'Hello'), (2,3.,"World")],
- ...              dtype=[('foo', 'i4'),('bar', 'f4'), ('baz', 'S10')])
+ >>> x = np.array([('Rex', 9, 81.0), ('Fido', 3, 27.0)],
+ ...              dtype=[('name', 'U10'), ('age', 'i4'), ('weight', 'f4')])
  >>> x
- array([(1, 2.0, 'Hello'), (2, 3.0, 'World')],
-      dtype=[('foo', '>i4'), ('bar', '>f4'), ('baz', '|S10')])
+ array([('Rex', 9, 81.0), ('Fido', 3, 27.0)],
+       dtype=[('name', 'S10'), ('age', '<i4'), ('weight', '<f4')])
 
-Here we have created a one-dimensional array of length 2. Each element of
-this array is a structure that contains three items, a 32-bit integer, a 32-bit
-float, and a string of length 10 or less. If we index this array at the second
-position we get the second structure: ::
+Here ``x`` is a one-dimensional array length 2, whose datatype is a structure
+with three fields: 1. A string of length 10 or less named 'name', 2. a 32-bit
+integer named 'age', and 3. a 32-bit float named 'weight'.
+
+If you index ``x`` at position 1 you get a structure::
 
  >>> x[1]
- (2,3.,"World")
+ ('Fido', 3, 27.0)
 
-Conveniently, one can access any field of the array by indexing using the
-string that names that field. ::
+You can access and modify individual fields of a structured array by indexing
+with the field name::
 
- >>> y = x['bar']
- >>> y
- array([ 2.,  3.], dtype=float32)
- >>> y[:] = 2*y
- >>> y
- array([ 4.,  6.], dtype=float32)
+ >>> x['age']
+ array([9, 3], dtype=int32)
+ >>> x['age'] = 5
  >>> x
- array([(1, 4.0, 'Hello'), (2, 6.0, 'World')],
-       dtype=[('foo', '>i4'), ('bar', '>f4'), ('baz', '|S10')])
+ array([('Rex', 5, 81.0), ('Fido', 5, 27.0)],
+       dtype=[('name', 'S10'), ('age', '<i4'), ('weight', '<f4')])
+
+Structured arrays are designed for low-level manipulation of structured data,
+for example for interpreting binary blobs. Structured datatypes are designed to
+mimic 'structs' in the C language, making them useful for interfacing with C
+code. For these purposes numpy supports specialized features such as subarrays
+and nested datatypes, and allows manual control over the memory layout of the
+structure.
+
+If you only wish to manipulate tabular data with labelled columns, you are
```

Comment:
Maybe "For simple manipulation of tabular data, other pydata projects, such as pandas, xarray, or DataArray, provide higher-level interfaces that may be more suitable."